apps/web/components/Shell.tsx

```diff
@@ -84,11 +85,7 @@ function useRedirectToOnboardingIfNeeded() {
   const query = useMeQuery();
   const user = query.data;
 
-  const [isRedirectingToOnboarding, setRedirecting] = useState(false);
-
-  useEffect(() => {
-    user && setRedirecting(shouldShowOnboarding(user));
-  }, [router, user]);
+  const isRedirectingToOnboarding = user && shouldShowOnboarding(user);
```

Comment:
I believe state is unnecessary here as any change in `query` will cause a rerender and `isRedirectingToOnboarding` can be considered a derived or computed state from `query.data`